azure-client-runtime/src/main/java/com/microsoft/azure/AzureServiceClient.java

```diff
@@ -49,6 +49,28 @@ protected AzureServiceClient(String baseUrl, ServiceClientCredentials credential
      */
     protected AzureServiceClient(RestClient restClient) {
         super(restClient);
+
+        final ServiceClientCredentials credentials = restClient().credentials();
+        final RxNettyAdapter rxnClient = new RxNettyAdapter();
+        httpClient = new RequestPolicyChain(new CredentialsPolicy.Factory(credentials), new RequestPolicy.Factory() {
```

Comment:
I'm curious about the name RequestPolicyChain, but I'll keep going through this review to see if it makes more sense.